gems/sorbet-runtime/lib/types/private/methods/signature_validation.rb

```diff
@@ -199,6 +202,7 @@ def self.validate_override_types(signature, super_signature)
     super_signature.kwarg_types.each do |name, super_type|
       type = signature.kwarg_types[name]
       if !super_type.subtype_of?(type)
+        next if TypeVariable === type || TypeVariable === super_type
```

Comment:
Could you add a test with `Array[Elem]` as return type?
I'd expect it to fail

Reply:
Does d3321592f look right?

Reply:
Sorry, I wasn't clear. Could you add the following test?
```
module AbstractGenericModule
  extend T::Generic
  extend T::Sig
  interface!

  C = type_member
  sig {abstract.returns(T::Array[C])}
  def hi; end
end

class Foo
  extend T::Generic
  extend T::Sig
  include AbstractGenericModule

  C = type_member(fixed: String)

  sig {implementation.returns(T::Array[String])}
  def hi
    "hi"
  end
end
```

Reply:
4250990 Ah, right. The current implementations fails in this case. Do you think it should be fixed? I'm not sure if there is a good way to fix it without changing `subtype?`, though...